src/docs/release/breaking-changes/platform-views-using-html-slots-web.md

```diff
@@ -0,0 +1,196 @@
+changes often result in DOM operations.
+
+Flutter web used to render its Platform Views (`HtmlElementView` widgets)
+directly into its corresponding position of the DOM.
+
+Using certain DOM elements as the "target" of some DOM operations causes those
+elements to lose their internal state. In practice this means that `iframe` tags
+are going to reload, `video` players might restart, or a form being edited might
+lose the edits made to it.
+
+This change was made to:
+
+* make Platform Views behave in a stable way in Flutter web.
+* unify the way Platform Views were rendered in the web for both rendering
+backends (`html` and `canvaskit`).
+
+## Description of change
+
+A Flutter web app is now rendered inside a common Shadow Root in which slot
+elements are used to represent platform views. The actual content of each
+platform view is rendered *outside* of the Shadow Root, as a child of the Shadow
+Root host.
+
+### Before
+
+<!-- skip -->
+```html
+...
+
+<flt-glass-pane>
+  ...
+  <div id="platform-view">Contents</div> <!-- canvaskit -->
+  <!-- OR -->
+  <flt-platform-view>
+    #shadow-root
+    | <div id="platform-view">Contents</div> <!-- html -->
+  </flt-platform-view>
+  ...
+</flt-glass-pane>
+
+...
+```
+
+### After
+
+<!-- skip -->
+```html
+...
+
+<flt-glass-pane>
+  #shadow-root
+  | ...
+  | <flt-platform-view-slot>
+  |   <slot name="platform-view-1" />
+  | </flt-platform-view-slot>
+  | ...
+  <flt-platform-view slot="platform-view-1">
+    <div id="platform-view">Contents</div>
+  </flt-platform-view>
+  ...
+</flt-glass-pane>
+
+...
+```
+
+Now, when the framework needs to move DOM nodes around, it'll operate over
+`flt-platform-view-slot`s, which only contain the `slot` tag that "projects" the
+content defined in the `flt-platform-view` tags outside of the Shadow Root
+(which never move), thus preventing the reloads.
+
+From an app's perspective, this change is transparent, **however** this was
+deemed a _breaking change_ because tests that make assumptions about the DOM of
+a Flutter web app can break.
+
+## Migration guide
+
+### Code
+
+The content returned by the users' Platform View Factory used to be resized and
+positioned by the framework, but not anymore. Ensure that your returned
+`html.Element` takes the whole space allocated for it by the framework, by
+setting its `style.width` and `style.height` properties to `'100%'`:
+
+<!-- skip -->
+```dart
+ui.platformViewRegistry.registerViewFactory(viewType, (int viewId) {
+  final html.Element htmlElement = html.DivElement()
+    // ..other props
+    ..style.width = '100%'
+    ..style.height = '100%';
+  // ...
+  return htmlElement;
+});
+```
+
+Otherwise the engine will print a warning message to the console similar to:
+
+<!-- skip -->
+```bash
+Height of Platform View type: [$viewType] may not be set. Defaulting to `height: 100%`.
+Set `style.height` to any appropriate value to stop this message.
```

Comment:
Hmm, do we need this? Why can't I use relative layout (flexbox, etc) to fill the space?

Reply:
If you use any other way of filling the space, you need to set height/width to `auto`, or anything that is not empty. In the engine the only thing we look at is that width/height is set.

Reply:
I'll add a "But I don't want to set width: 100% in my platform view" section